hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieMergeHandle.java

```diff
@@ -192,10 +192,17 @@ private void init(String fileId, String partitionPath, HoodieBaseFile baseFileTo
       writeStatus.getStat().setFileId(fileId);
       setWriteStatusPath();
 
-      // Create Marker file,
+      // If this is a second or subsequent attempt to create the data file, try to recover existing version.
+      if (recoverWriteStatusIfAvailable(partitionPath,FSUtils.makeBaseFileName(this.instantTime, this.writeToken,
+          this.fileId, hoodieTable.getBaseFileExtension()), this.instantTime)) {
```

Comment:
Not sure whether we can do this, is the `writeToken` in random format for multiple attempts from spark or it is always the same?

Reply:
Write token is created with Spark (partitionID, stageId, taskID) tuple.  For creating the same data file, if two executors are making an attempt, the fileId and commit timestamp portions will be the same.  However, the taskID in the write token is guaranteed to be unique.

Reply:
So how to reuse the write status because the file name already changes, and for Flink we even have a eager flush(one instant would flush multiple files in one filegroup).